onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -156,6 +156,30 @@ CUDAExecutionProvider::PerThreadContext::~PerThreadContext() {
   } catch (const std::exception& ex) {
     LOGS_DEFAULT(ERROR) << "cudnnDestroy threw:" << ex.what();
   }
+
+  if (!external_stream_ && stream_) {
+    CUDA_CALL(cudaStreamDestroy(stream_));
+  }
+}
+
+Status CUDAExecutionProvider::PerThreadContext::SetComputeStream(void* stream) {
+  if (stream != stream_) {
+    if (stream_) {
+      CUDA_RETURN_IF_ERROR(cudaStreamDestroy(stream_));
```

Comment:
What if the stream is not owned ? Can it be destroyed by the PerThreadContext ?

Reply:
Good question. I followed the logic [here](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/cuda/cuda_execution_provider.cc#L352-L360). I think for the external stream, it should be not destroyed by the PerThreadContext. If you also agree, I will make a change here.